packages/composites/proof-of-reserves/test/integration/adapter.test.ts

```diff
@@ -0,0 +1,98 @@
+import * as process from 'process'
+import { AdapterRequest } from '@chainlink/types'
+import request, { SuperTest, Test } from 'supertest'
+import { server as startServer } from '../../src'
+import { mockPoRindexerSuccess, mockEthBalanceSuccess } from './fixtures'
+import * as nock from 'nock'
+import { AddressInfo } from 'net'
+
+beforeAll(() => {
+  if (process.env.RECORD) {
+    nock.recorder.rec()
+  }
+  process.env.ETH_BALANCE_ADAPTER_URL = 'https://eth-balance-adapter.com'
```

Comment:
When modifying environment variables we should ensure that there are no side effects by saving the current env and restoring it at the end.

See [other integration tests](https://github.com/smartcontractkit/external-adapters-js/blob/develop/packages/composites/anchor/test/integration/prices/beth.test.ts#L88) as an example.